src/giftCards/GiftCardCreateDialog/GiftCardCreateDialogMoneyInput.tsx

```diff
@@ -0,0 +1,87 @@
+import TextWithSelectField from "@saleor/components/TextWithSelectField";
+import { ChangeEvent } from "@saleor/hooks/useForm";
+import useLocalStorage from "@saleor/hooks/useLocalStorage";
+import { mapSingleValueNodeToChoice } from "@saleor/utils/maps";
+import * as React from "react";
+import { useEffect } from "react";
+import { useIntl } from "react-intl";
+
+import { getGiftCardErrorMessage } from "../GiftCardUpdate/messages";
+import { GiftCardCreateFormData } from "./GiftCardCreateDialogForm";
+import { giftCardCreateDialogMessages as messages } from "./messages";
+import { useChannelCurrencies } from "./queries";
+import { useGiftCardCreateDialogFormStyles as useStyles } from "./styles";
+import { GiftCardCreateFormCommonProps } from "./types";
+
+interface GiftCardCreateDialogMoneyInputProps
+  extends GiftCardCreateFormCommonProps {
+  set: (data: Partial<GiftCardCreateFormData>) => void;
+}
+
+const GiftCardCreateDialogMoneyInput: React.FC<GiftCardCreateDialogMoneyInputProps> = ({
+  errors,
+  data: { balanceAmount, balanceCurrency },
+  change,
+  set
+}) => {
+  const intl = useIntl();
+  const classes = useStyles({});
+
+  const { data: channelCurrenciesData } = useChannelCurrencies({});
+
+  const { channelCurrencies } = channelCurrenciesData?.shop;
+
+  const [savedCurrency, setCurrency] = useLocalStorage(
```

Comment:
Pass it from view level

Reply:
Do you mean from `GiftCardCreateDialogForm`? Is there a benefit to this? I think that encapsulated component like this is better than doing it in the view level just to pass it down.